designs/2019-pass-cwd-from-cli-engine/README.md

```diff
@@ -0,0 +1,66 @@
+- Start Date: 2019-07-25
+- RFC PR: https://github.com/eslint/rfcs/pull/35
+- Authors: Eric Wang
+
+# Pass cwd from cli engine
+
+## Summary
+Make the CLIEngine `cwd` option available to rules to avoid the misalignment between it and process.cwd()
+
+see: https://github.com/eslint/eslint/issues/11218
+
+## Motivation
+To avoid the misalignment between it and `process.cwd()`.
+There is a situation where a rule need to get the relative path of the current file to the root folder of the project.
+But it's hard to guarantee that all developers will open the root folder in their IDE (VSCode with eslint plugin for example), especially when the project grows.
+
+Use case:
+Say, the project is supposed to be open in `web` folder.
+But as the project grows, people tends to open some subfolder under the `web` to minimize sidebar.
+`eslint` plugin in VSCode will then run the `eslint` from the subfolder which cause the `process.cwd` not to return the expected value (e.g. the `web` folder)
+
+Expected:
+It is expected that the rule has access to the `cwd` in the option of the CLI engine.
+
+## Detailed Design
+### Change in Linter
+Modify the `Linter` constructor to accept an object with a nullable string parameter `cwd`.
+```
+const linter = new Linter({
+  cwd: '/path/to/cwd'
+});
+```
+or `const linter = new Linter({})`.
+
+### Change in Context
+This RFC adds `getCwd()` method to rule context.
+The `getCwd()` method returns the value of the `cwd` option that was given in `Linter` constructor.
+If the `cwd` is `undefined` in the `Linter` but the `process` exists, `getCwd()` will return `process.cwd()`.
```

Comment:
Thank you for your work.

As my last concern, I think that we should move this fallback behavior to the `Linter` constructor. Because in the current spec, if a user changed the current working directory after they instantiate `Linter` object, the value of `context.getCwd()` will change. This behavior is different from the other classes in ESLint. It's better if the classes have the same manner.